packages/core/components/Modal/CodeSnippet/ProcessFilesCodeSnippetModal.tsx

```diff
@@ -0,0 +1,110 @@
+import * as React from "react";
+import { useDispatch, useSelector } from "react-redux";
+
+import { ModalProps } from "..";
+import { selection, interaction } from "../../../state";
+import { setProcessFilesPythonSnippet } from "../../../state/interaction/actions";
+import CodeSnippet from ".";
+
+export default function ProcessFiles({ onDismiss }: ModalProps) {
+    const dispatch = useDispatch();
+    const fileSelection = useSelector(selection.selectors.getFileSelection);
+    const snippet = useSelector(interaction.selectors.getProcessFilesPythonSnippet);
+
+    React.useEffect(() => {
+        let mounted = true;
+
+        (async () => {
+            try {
+                const details = await fileSelection.fetchAllDetails();
+                if (!mounted) return;
+
+                const safe = (s: string) =>
+                    String(s ?? "")
+                        .replace(/\\/g, "\\\\")
+                        .replace(/"/g, '\\"');
+
+                // Plugin registry
+                const EXT_TO_PKG: Array<[string, string]> = [
```

Comment:
minor: since this is the exact same registry as ConvertToZarr, wondering if this and the extension mapping code should actually just be shared between them as a util somehow